addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/ParamMinerPanel.java

```diff
@@ -19,41 +19,162 @@
+            ParamGuesserScanController scanController,
+            ParamMinerOptions options,
+            Runnable scanStartRunnable) {
+        super("paramminer", ExtensionParamMiner.getIcon(), scanController);
+
+        getNewScanButton().setText(Constant.messages.getString("paramminer.toolbar.button.new"));
+        getNewScanButton().setIcon(getIcon());
+        getNewScanButton().addActionListener(e -> scanStartRunnable.run());
+
+        this.options = options;
+    }
+
+    @Override
+    protected Component getWorkPanel() {
+        if (mainPanel == null) {
+            mainPanel = new JPanel(new BorderLayout());
+
+            emptyTableModel = new ParamMinerHistoryTableModel();
+            historyTable = new ParamMinerResultsTable(emptyTableModel);
+            outputArea = new JTextPane();
+            outputArea.setEditable(false);
+
+            tabbedPane = new JTabbedPane();
+            tabbedPane.addTab(
+                    Constant.messages.getString("paramminer.panel.tab.history"),
+                    new JScrollPane(historyTable));
+            tabbedPane.addTab(
+                    Constant.messages.getString("paramminer.panel.tab.output"),
+                    new JScrollPane(outputArea));
+            mainPanel.add(tabbedPane);
+        }
+        return mainPanel;
+    }
+
+    @Override
+    public void scannerStarted(GuesserScan scan) {
+        super.scannerStarted(scan);
+
+        scan.addProgressListener(getProgressListener());
+    }
+
+    @Override
+    protected void switchView(GuesserScan scan) {
+        if (scan != null) {
+            historyTable.setModel(scan.getTableModel());
+            outputArea.setText(scan.getOutput());
+        } else {
+            historyTable.setModel(emptyTableModel);
+            outputArea.setText("");
+        }
+        mainPanel.revalidate();
+        mainPanel.repaint();
+    }
+
+    @Override
+    protected JButton getNewScanButton() {
+        if (startScanButton == null) {
+            startScanButton = new JButton();
+        }
+        return startScanButton;
+    }
+
+    @Override
+    protected int getNumberOfScansToShow() {
+        return options.getMaxFinishedScansInUi();
+    }
+
+    @Override
+    public void clearFinishedScans() {
+        if (options.isPromptToClearFinishedScans()) {
+            JCheckBox dontPromptCheckBox =
+                    new JCheckBox(
+                            Constant.messages.getString(
+                                    "paramminer.toolbar.confirm.clear.dontPrompt"));
+            Object[] messages = {
+                Constant.messages.getString("paramminer.toolbar.confirm.clear"),
+                "\n",
+                dontPromptCheckBox
+            };
+            int option =
+                    JOptionPane.showConfirmDialog(
+                            View.getSingleton().getMainFrame(),
+                            messages,
+                            Constant.PROGRAM_NAME,
```

Comment:
Should this use a param miner string so that it's more clear what the dialog applies to?